package com.sachin.Student;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import org.springframework.cloud.client.loadbalancer.LoadBalanced;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.web.client.RestTemplate;
import org.springframework.web.reactive.function.client.WebClient;

@SpringBootApplication
@EnableDiscoveryClient
@ComponentScan(basePackages = "com.sachin")
public class StudentApplication
{
	public static void main(String[] args)
	{
		SpringApplication.run(StudentApplication.class, args);
	}

	@Bean
	@LoadBalanced
	WebClient.Builder webClientBuilder()
	{
		return WebClient.builder();
	}

	@Bean
	@LoadBalanced
	RestTemplate restTemplate()
	{
		return new RestTemplate();
	}
}
